src/IntegrationTests/CustomMapFrom/MapObjectPropertyFromSubQuery.cs

```diff
@@ -689,27 +1007,43 @@ public class DatabaseInitializer : DropCreateDatabaseAlways<ClientContext>
         protected override void Seed(ClientContext context)
         {
             var product = context.Products.Add(new Product { ECommercePublished = true, Articles = new[] { new Article { IsDefault = true, NationId = 1, ProductId = 1 } } });
-            context.ProductArticles.Add(new ProductArticle { Products = new[] { product.Entity } });
+            context.ProductArticles.Add(new ProductArticleA { Name = "P1", A = "a", Products = new[] { product.Entity } });
+            
+            product = context.Products.Add(new Product { ECommercePublished = true, Articles = new[] { new Article { IsDefault = true, NationId = 1, ProductId = 1 } } });
+            context.ProductArticles.Add(new ProductArticleB { Name = "P2", B = "b", Products = new[] { product.Entity } });
+
+            product = context.Products.Add(new Product { ECommercePublished = true, Articles = new[] { new Article { IsDefault = true, NationId = 1, ProductId = 1 } } });
+            context.ProductArticles.Add(new ProductArticle { Name = "P3", Products = new[] { product.Entity } });
         }
     }
 
     public class ClientContext : LocalDbContext
     {
         public DbSet<Product> Products { get; set; }
         public DbSet<ProductArticle> ProductArticles { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+            modelBuilder.Entity<ProductArticleA>();
+            modelBuilder.Entity<ProductArticleB>();
+        }
     }
 }
-
 public class MapObjectPropertyFromSubQueryWithCollectionSameName : NonValidatingSpecBase, IAsyncLifetime
 {
     protected override MapperConfiguration CreateConfiguration() => new(cfg =>
     {
-        cfg.CreateProjection<ProductArticle, ProductArticleModel>();
-        cfg.CreateProjection<Product, ProductModel>()
+        cfg.CreateMap<ProductArticle, ProductArticleModel>()
+            .Include<ProductArticleA, ProductArticleAModel>()
+            .Include<ProductArticleB, ProductArticleBModel>();
+        cfg.CreateMap<ProductArticleA, ProductArticleAModel>();
+        cfg.CreateMap<ProductArticleB, ProductArticleBModel>();
```

Comment:
Similar to `IncludeMembers`, at least a few of these tests should have some configuration in the derived map. Certainly a subquery using the let clause and perhaps some random `MapFrom` to see that it's used.